docs/source/extensions/datamodules.rst

```diff
@@ -326,15 +326,35 @@ on_after_batch_transfer
 on_load_checkpoint
 ~~~~~~~~~~~~~~~~~~
 
+.. warning:: ``on_load_checkpoint`` is deprecated in v1.6 and will be removed in v1.8.
+    Please use ``load_state_dict`` instead.
+
+
 .. automethod:: pytorch_lightning.core.datamodule.LightningDataModule.on_load_checkpoint
     :noindex:
 
+load_state_dict
+~~~~~~~~~~~~~~~~~~
+
+.. automethod:: pytorch_lightning.core.datamodule.LightningDataModule.load_state_dict
+    :noindex:
+
 on_save_checkpoint
 ~~~~~~~~~~~~~~~~~~
 
+.. warning:: ``on_save_checkpoint`` is deprecated in v1.6 and will be removed in v1.8.
```

Comment:
Instead, I would not mention the deprecated methods in the docs.

Also, have you grepped the docs directory for any occurrences?

Reply:
> Also, have you grepped the docs directory for any occurrences?

Yep, this is it for datamodules. Will have a separate overall docs PR for `checkpointing.rst`